### DISCUSSION TITLE
Welcome to medusa-strapi-repo Discussions!

### DISCUSSION THREAD

**Discussion by SGFGOV** [SGFGOV/medusa-strapi-repo#46] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by l-shirinyan** [comment]
Hi @SGFGOV how I can fix this 
![quest](https://github.com/user-attachments/assets/96724129-2a3a-4fb5-9e9e-7db00f77c5e3)

**Comment by SGFGOV** [comment]
Well you need to edit the medusa strapi plugin

**Comment by l-shirinyan** [comment]
@SGFGOV the problem two way sync working correctly , so If I adding product from medusa it was also adding in strapi but problem when I  checking strapi-plugin-medusa gettig error described in screen

**Reply by SGFGOV** [reply]
Check in content manager

**Reply by l-shirinyan** [reply]
@SGFGOV in content manager everything adding correctly,but theres also lot of bugs ,example if in strapi
 I delete some of products or publishing some of products other products are deleting.

**Comment by l-shirinyan** [comment]
const { getDefaultRoleAssumerWithWebIdentity } = require('@aws-sdk/client-sts');
const { defaultProvider } = require('@aws-sdk/credential-provider-node');
const { config } = require('process');

const strapiOptions = 
    {
        sync_on_init:true,
        encryption_algorithm: "aes-256-cbc",
        strapi_protocol: process?.env?.STRAPI_PROTOCOL,
        strapi_default_user: {
            username: process?.env?.STRAPI_MEDUSA_USER,
            password: process?.env?.STRAPI_MEDUSA_PASSWORD,
            email: process?.env?.STRAPI_MEDUSA_EMAIL,
            confirmed: true,
            blocked: false,
            provider: "local"
        },
        strapi_host: process?.env?.STRAPI_SERVER_HOSTNAME,
        strapi_admin: {
            username:
                process?.env?.STRAPI_SUPER_USERNAME ||
                "SuperUser",
            password:
                process?.env?.STRAPI_SUPER_PASSWORD ||
                "MedusaStrapi1",
            email:
                process?.env?.STRAPI_SUPER_USER_EMAIL ||
                "support@medusa-commerce.com"
        },
        strapi_port: process?.env?.STRAPI_PORT,
        strapi_secret: process?.env?.STRAPI_SECRET,
        strapi_public_key: process?.env?.STRAPI_PUBLIC_KEY,
        strapi_ignore_threshold: 3
    }


const providerConfigAws = (env) => {
	return {
		provider: 'aws-s3',
		sizeLimit: 250 * 1024 * 1024,
		providerOptions: {
			accessKeyId: env('AWS_ACCESS_KEY_ID'),
			secretAccessKey: env('AWS_ACCESS_SECRET'),
			credentialDefaultProvider: !env('AWS_ACCESS_KEY_ID')
				? defaultProvider({
						roleAssumerWithWebIdentity: getDefaultRoleAssumerWithWebIdentity,
				  })
				: undefined,
			region: env('S3_REGION'),
			params: {
				Bucket: env('S3_BUCKET'),
			},
		},
		actionOptions: {
			upload: {},
			uploadStream: {},
			delete: {},
		},
		breakpoints: {
			xlarge: 1920,
			large: 1000,
			medium: 750,
			small: 500,
			xsmall: 64,
		},
	};
};

const providerConfigLocal = {
	providerOptions: {
		localServer: {
			maxage: 300000,
		},
	},
};

module.exports = ({ env }) => ({
	// ...
	seo: {
		enabled: true,
	},
	// ...
	'strapi-plugin-medusajs': true,
	// 'strapi-plugin-medusajs': {
	// 	enabled: true,
	// 	config: {
	// 		MEDUSA_STRAPI_SECRET: env('MEDUSA_STRAPI_SECRET'),
	// 	},
	// },
	'users-permissions': {
		enabled: true,
		config: {
			jwt: {
				expiresIn: '1h',
			},
			jwtSecret: env('JWT_SECRET', 'test-jwt'),
			ratelimit:
				process.env.NODE_ENV == 'test'
					? {
							interval: 60000,
							max: 100000,
					  }
					: {
							headers: true,
					  },
		},
	},
	'strapi-plugin-sso': {
		enabled: true,
		config: {
			MEDUSA_SERVER: env('MEDUSA_BACKEND_URL', 'http://localhost:9000'),
			MEDUSA_ADMIN: env('MEDUSA_BACKEND_ADMIN', 'http://localhost:7000'),
			MEDUSA_STRAPI_SECRET: env('MEDUSA_STRAPI_SECRET', 'no_secret'),
		},
	},

	upload: {
		enabled: true,
		config: env('NODE_ENV') == 'test' ? providerConfigLocal : providerConfigAws(env),
	},
	email: {
		enabled: true,
		config: {
			provider: 'sendgrid',
			providerOptions: {
				apiKey: env('SENDGRID_API_KEY'),
			},
			settings: {
				defaultFrom: env('SENDGRID_NOTIFICATION_FROM_ADDRESS'),
				defaultReplyTo: env('SENDGRID_NOTIFICATION_FROM_ADDRESS'),
			},
		},
	},
	sentry: {
		enabled: true,
		config: {
			dsn: env('SENTRY_DSN'),
			sendMetadata: true,
		},
	},
	meilisearch: {
		enabled: true,
		config: {
			// Your meili host
			host: env('MEILISEARCH_HOST'),
			// Your master key or private key
			apiKey: env('MEILISEARCH_MASTER_KEY'),
		},
	},
	'content-versioning': {
		enabled: true,
	},
	'generate-data': {
		enabled: true,
	},

	'strapi-google-translator': {
		enabled: true,
		config: {
			backendUrl: env('STRAPI_BACKEND_URL'),
			apiToken: env('STRAPI_GOOGLE_TRANSLATE_API_TOKEN'),
			googleJson: env('GOOGLE_TRANSLATE_JSON'),
		},
		'strapi-plugin-sitemap': {
			enabled: true,
		},
	},
	comments: {
		enabled: true,
		config: {
			badWords: true,
			moderatorRoles: ['Authenticated'],
			approvalFlow: ['api::page.page'],
			entryLabel: {
				'*': ['Title', 'title', 'Name', 'name', 'Subject', 'subject'],
				'api::product.product': ['name'],
			},
			reportReasons: {
				MY_CUSTOM_REASON: 'MY_CUSTOM_REASON',
			},
			gql: {
				auth: true,
			},
		},
	},
	io: {
		enabled: true,
		config: {
			IOServerOptions: {
				cors: { origin: env('STORE_CORS'), methods: ['GET'] },
			},
			contentTypes: '*',
			events: [
				{
					name: 'connection',
					handler: ({ strapi }, socket) => {
						strapi.log.info(`[io] new connection with id ${socket.id}`);
					},
				},
			],
		},
	},
	resolve: "medusa-plugin-strapi-ts",
	options: {
		...strapiOptions
	}
});
here is my configs from plugins.js file
